run_checkm_bakta.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+#SBATCH -J 500_ncbi_redgenes
+#SBATCH -p short
+#SBATCH -t 20:00:00
+#SBATCH -N 1
+#SBATCH -c 4
+#SBATCH --mem 64g
+#SBATCH -o /panfs/roles/redgenes/slurm-%x-%A-%a-%j-%N.out
+#SBATCH -e /panfs/roles/redgenes/slurm-%x-%A-%a-%j-%N.err
+#SBATCH --export ALL
+#SBATCH --mail-type=ALL,TIME_LIMIT_50,TIME_LIMIT_90
+#SBATCH --mail-user=roles@health.ucsd.edu
+
+set -x
+set -e
+set -o pipefail
+
+source /home/roles/anaconda/bin/activate
+conda activate checkm_bakta
+
+# Check input parameters
+if [ "$#" -ne 3 ]; then
+    echo "Usage: $0 <input_dir_with_fasta_files> <output_dir> <number_of_parallel_jobs>"
+    exit 1
+fi
+
+input_dir=$1
+output_dir=$2
+parallel_jobs=$3
+
+# Create output directory if it doesn't exist
+mkdir -p "$output_dir"
+
+# Function to process each FASTA file
+process_fasta() {
+    fasta_file=$1
+    base_name=$(basename "$fasta_file" .fasta)
+
+    # Create directory for each fasta file in the specified output directory
+    sample_dir="$output_dir/$base_name"
+    mkdir -p "$sample_dir/checkm"
+    mkdir -p "$sample_dir/bakta"
+
+    # Run CheckM
+    checkm lineage_wf --tab_table -x fasta "$fasta_file" "$sample_dir/checkm" > "$sample_dir/checkm/lineage.log"
+    if [ $? -ne 0 ]; then
+        echo "CheckM failed for $fasta_file"
+        return 1
+    fi
+
+    # Extract completeness and contamination from lineage.log
+    completeness=$(awk '/Completeness/ {print $12}' "$sample_dir/checkm/lineage.log")
+    contamination=$(awk '/Contamination/ {print $13}' "$sample_dir/checkm/lineage.log")
+    if (( $(echo "$completeness > 95 && $contamination < 5" | bc -l) )); then
+        echo "PASS" > "$sample_dir/checkm_results.txt"
+    else
+        echo "FAIL" > "$sample_dir/checkm_results.txt"
+        return 1
+    fi
+
+    # If passed, run Bakta
+    if [ "$(cat "$sample_dir/checkm_results.txt")" == "PASS" ]; then
+        bakta --cpus 4 --outdir "$sample_dir/bakta" --prefix "$base_name" "$fasta_file"
+        if [ $? -ne 0 ]; then
+            echo "Bakta annotation failed for $fasta_file"
+            echo "FAIL" > "$sample_dir/bakta_results.txt"
+            return 1
+        else
+            echo "PASS" > "$sample_dir/bakta_results.txt"
+        fi
+    fi
+}
+
+export -f process_fasta
+
+# Find all fasta files and process them in parallel, but serially per file for CheckM and Bakta
+find "$input_dir" -name "*.fasta" -print0 | parallel -0 -j $parallel_jobs -I{} bash -c 'process_fasta "{}"'
```

Comment:
This parallel seems to be taking empty $fasta_file?
```
Processing checkM results...
mkdir: cannot create directory ‘//checkm’: Permission denied
mkdir: cannot create directory ‘//bakta’: Permission denied
mkdir: cannot create directory ‘//ko’: Permission denied
/projects/greengenes2/20240331_troubleshoot_kofamscan/data_ecoli/GCF_000005845.2_ASM584v2_genomic.fna: line 11: //checkm/lineage.log: No such file or directory
```
See if you can reproduce this error by running 
1) `chmod 700 run_checkm_bakta.sh` if the bash file is currently not an excutable
2) change the part of finding fna files to `find "$input_dir" -type f \( -name "*.fasta" -o -name "*.fna" -o -name "*.fa" \) -print0` so it recognizes different extension.
3) `./run_checkm_bakta.sh /projects/greengenes2/20240331_troubleshoot_kofamscan/data_ecoli ./ 4`

I can see correct filenames are being piped into the parallel part by doing 
```
find "$input_dir" -type f \( -name "*.fasta" -o -name "*.fna" -o -name "*.fa" \) -print0 |
while IFS= read -r -d '' file; do
    echo "$file"
    process_fasta "$file"
done
```
which gives
`/projects/greengenes2/20240331_troubleshoot_kofamscan/data_ecoli/GCF_000005845.2_ASM584v2_genomic.fna`
but not sure why process_fasta is taking in empty file paths.